frappe/test_runner.py

```diff
@@ -73,7 +74,9 @@ def main(app=None, module=None, doctype=None, verbose=False, tests=(),
 		else:
 			ret = run_all_tests(app, verbose, profile, ui_tests, failfast=failfast, junit_xml_output=junit_xml_output)
 
-		frappe.utils.scheduler.enable_scheduler()
+		if cint(frappe.db.get_value("System Settings", fieldname="enable_scheduler")):
+			frappe.utils.scheduler.disable_scheduler()
+			if frappe.db: frappe.db.commit()
```

Comment:
@e-lobo Isn't this redundant?

Reply:
@assemmarwan this is to ensure any write operations in db are completed before tests start to run.. there is an issue if we remove frappe.db.commit() when we run tests with --doctype

Reply:
@assemmarwan fixed.. it was in the wrong place 👍